src/insights_client/tests/test_metrics.py

```diff
@@ -34,27 +35,92 @@ def factory(config):
 @fixture
 def rhsm_config_file_factory():
     def factory(**config):
+        # keep these default values initialized
+        if "hostname" not in config:
+            config["hostname"] = "subscription.rhsm.redhat.com"
+        if "port" not in config:
+            config["port"] = "443"
         config = u"""[server]
-hostname = cert-api.access.redhat.com
-port = 443
-proxy_hostname = %s
-proxy_port = %s
-proxy_user = %s
-proxy_password = %s
-
+%s
+%s
+%s
+%s
+%s
+%s
 [rhsm]
-repo_ca_cert =
-consumerCertDir =
-""" % (
-            config.get("proxy_hostname", ""),
-            config.get("proxy_port", ""),
-            config.get("proxy_user", ""),
-            config.get("proxy_password", ""),
-        )
+%s
+%s
+""" % tuple(
+            "%s = %s" % (key, config[key]) if key in config else ""
+            for key in ("hostname", "port", "proxy_hostname", "proxy_port", "proxy_user", "proxy_password", "repo_ca_cert", "consumerCertDir"))
         return _tempfile(config)
 
     return factory
 
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_missing_server_section(proxy_settings, config_file_factory):
```

Comment:
This case tests two things at the same time:

- Correct handling of a missing _server_ section.
- Recognition of the _BASIC_ authmethod.

It’s a candidate for a split.